### PR TITLE
Detect version of sed in Makefile

### PR DESCRIPTION
**Pull request by silverwind**
It's possible to install GNU sed on Darwin or FreeBSD so it's better to not assume BSD sed on those platforms but to instead perform version detection for the `GNU` string and only use BSD syntax if absent.

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lafriks** [comment]
🚀

**Review by techknowlogick — CHANGES_REQUESTED** [review]
Not sure this is the way to go. sed --version fails on macOS

**Comment by silverwind** [comment]
Yeah, BSD `sed` knows no `--version` argument but it's no issue as long as there's no `GNU` string in it.

**Comment by silverwind** [comment]
Switching to `perl` might be an option but I'm not totally certain that the regex engines are compatible and I'm not sure whether `sed` or `perl` is more likely to exist on a given system.

File: Makefile [hunk 1]
```diff
@@ -42,18 +41,16 @@ ifeq ($(HAS_GO), GO)
...
-	UNAME_S := $(shell uname -s)
-	ifeq ($(UNAME_S),Darwin)
-		SED_INPLACE := sed -i ''
-	endif
-	ifeq ($(UNAME_S),FreeBSD)
-		SED_INPLACE := sed -i ''
-	endif
+endif
+
+ifeq ($(shell sed --version | grep -q GNU && echo gnu),gnu)
```
Comment (techknowlogick) [inline Makefile]:
```suggestion
ifeq ($(shell sed --version 2>/dev/null | grep -q GNU && echo gnu),gnu)
```

This works on macOS as it suppresses error output and selects appropriate `SED_INPLACE`, but getting tested on non-macOS would be a good idea too.

Comment (jolheiser) [reply Makefile]: This seems to work on Windows using Git Bash. I didn't run in the Makefile, but this should be the same outcome.
sed-i

Comment (silverwind) [reply Makefile]: Did you see the error output during `make`? I assumed `shell` already suppresses output but I'm happy to add this if it helps.

Comment (techknowlogick) [reply Makefile]: yup, I suspect while `shell` suppresses output, it does it only for `stdout` whereas the suggestion above suppresses `stderr`

here is a tiny makefile example to test:
```
.PHONY: all
all:

ifeq ($(shell sed --version | grep -q GNU && echo gnu),gnu)
	@echo "yes"
else
	@echo "no"
endif
```

output on macOS
```
> make                                                                                                                                                                                            
sed: illegal option -- -
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
no
```

With the suggested change, the output becomes:
```
> make 
no
```

Comment (silverwind) [reply Makefile]: Weird, all versions of Make I use also suppress stderr, even 3.81 on macOS:

```
GNU Make 4.3
Built for x86_64-apple-darwin19.2.0
```
```
GNU Make 4.3
Built for x86_64-pc-linux-gnu
```
```
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
```

Anyways, it seems its an issue on some make versions so I'll add it.

Comment (techknowlogick) [reply Makefile]: My make reports version
```
> make -v
GNU Make 3.81
This program built for i386-apple-darwin11.3.0
```

Do you manage your make binary via homebrew?

Either way I'll revoke my request for change. Thanks for your work :)

**Review by techknowlogick — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (silverwind) [reply Makefile]: Oh I'm just dumb. All systems I use have GNU sed so it never ran in the error condition. LGTM.

**Pull request merged**